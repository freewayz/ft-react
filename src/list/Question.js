import React from 'react'
import {Link} from 'react-router-dom'

const Question = (question) => {
  const {title, id, library, reviewed, type, visibility} = question
  const editLink = `/edit/${id}`
  return (
    <div className='question__item'>
      <div className='question__item__col'> {id} </div>
      <div className='question__item__col'> {reviewed ? <i class='fas fa-check' /> : <i class='fas fa-check' />} </div>
      <div className='quetion__item__col question__title'>
        <a href={editLink}>
          {title}
        </a>
      </div>
      <div className='question__item__col'>{library}</div>
      <div className='question__item__col'>{visibility}</div>
      <div className='question__item__col'>{type}</div>
      <div><span role='img' aria-label='xml'>‚ùå</span></div>
      <div> <Link to={editLink}> Edit </Link> </div>
    </div>

  )
}

export default Question
