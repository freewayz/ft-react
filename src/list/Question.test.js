/* global expect jest it describe shallow */
import React from 'react'
import crateRouterContext from 'react-router-test-context'
import Question from './Question'

const questionProps = {
  title: 'Question Title',
  description: 'Question Description',
  id: 3
}

const context = crateRouterContext()
describe('Question Component', () => {
  it('create the right snapshots', () => {
    const tree = shallow(<Question {...questionProps} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders the Questions with the right texts', () => {
    const wrapper = shallow(<Question {...questionProps} />, {context})
    const divs = wrapper.find('div')
    // target the title div
    expect(divs.at(3).text()).toBe('Question Title')
    expect(divs.length).toBe(8)
  })
})
