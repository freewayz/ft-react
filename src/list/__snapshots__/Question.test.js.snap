// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Component create the right snapshots 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    description="Question Description"
    id={3}
    title="Question Title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="question__item__col"
>
           
          3
           
</div>,
        <div
          className="question__item__col"
>
           
          <i
                    class="fas fa-check"
          />
           
</div>,
        <div
          className="quetion__item__col question__title"
>
          <a
                    href="/edit/3"
          >
                    Question Title
          </a>
</div>,
        <div
          className="question__item__col"
/>,
        <div
          className="question__item__col"
/>,
        <div
          className="question__item__col"
/>,
        <div>
          <span
                    aria-label="xml"
                    role="img"
          >
                    ❌
          </span>
</div>,
        <div>
           
          <Link
                    replace={false}
                    to="/edit/3"
          >
                     Edit 
          </Link>
           
</div>,
      ],
      "className": "question__item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            3,
            " ",
          ],
          "className": "question__item__col",
        },
        "ref": null,
        "rendered": Array [
          " ",
          3,
          " ",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            <i
              class="fas fa-check"
/>,
            " ",
          ],
          "className": "question__item__col",
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "class": "fas fa-check",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          " ",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="/edit/3"
>
            Question Title
</a>,
          "className": "quetion__item__col question__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Question Title",
            "href": "/edit/3",
          },
          "ref": null,
          "rendered": "Question Title",
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "question__item__col",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "question__item__col",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "question__item__col",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            aria-label="xml"
            role="img"
>
            ❌
</span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "xml",
            "children": "❌",
            "role": "img",
          },
          "ref": null,
          "rendered": "❌",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            <Link
              replace={false}
              to="/edit/3"
>
               Edit 
</Link>,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " Edit ",
              "replace": false,
              "to": "/edit/3",
            },
            "ref": null,
            "rendered": " Edit ",
            "type": [Function],
          },
          " ",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="question__item__col"
>
             
            3
             
</div>,
          <div
            className="question__item__col"
>
             
            <i
                        class="fas fa-check"
            />
             
</div>,
          <div
            className="quetion__item__col question__title"
>
            <a
                        href="/edit/3"
            >
                        Question Title
            </a>
</div>,
          <div
            className="question__item__col"
/>,
          <div
            className="question__item__col"
/>,
          <div
            className="question__item__col"
/>,
          <div>
            <span
                        aria-label="xml"
                        role="img"
            >
                        ❌
            </span>
</div>,
          <div>
             
            <Link
                        replace={false}
                        to="/edit/3"
            >
                         Edit 
            </Link>
             
</div>,
        ],
        "className": "question__item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              3,
              " ",
            ],
            "className": "question__item__col",
          },
          "ref": null,
          "rendered": Array [
            " ",
            3,
            " ",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              <i
                class="fas fa-check"
/>,
              " ",
            ],
            "className": "question__item__col",
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "class": "fas fa-check",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " ",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="/edit/3"
>
              Question Title
</a>,
            "className": "quetion__item__col question__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Question Title",
              "href": "/edit/3",
            },
            "ref": null,
            "rendered": "Question Title",
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "question__item__col",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "question__item__col",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "question__item__col",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              aria-label="xml"
              role="img"
>
              ❌
</span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "xml",
              "children": "❌",
              "role": "img",
            },
            "ref": null,
            "rendered": "❌",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              <Link
                replace={false}
                to="/edit/3"
>
                 Edit 
</Link>,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Edit ",
                "replace": false,
                "to": "/edit/3",
              },
              "ref": null,
              "rendered": " Edit ",
              "type": [Function],
            },
            " ",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
